### PR TITLE
MBS-11614: Allow merging releases when a medium is empty

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11614

If a release has two mediums, each with 1 track, and another has two mediums, one empty and one with 1 track, we should allow merging the empty one into the non-empty one. After all, that data doesn't mean the medium actually has 0 tracks, just that we don't have enough info about the tracks on that medium.
Merging the other way around (into the release with the 0-track medium) should still fail, but with a more explanatory error message.

### CODE REVIEW

**Comment by mwiencek** [comment]
> Merging the other way around (into the release with the 0-track medium) should still fail, but with a more explanatory error message.

The error doesn't really explain it to me. :) If the releases are the same, why shouldn't either direction be valid and have the same outcome for the empty medium?

**Comment by mwiencek** [comment]
I was thinking of it more like filling in a null value with a non-null one, rather than overwriting. (Like I wouldn't expect it to overwrite the medium format or name of the empty one.)

The two tracklisty -> one empty-tracklist merge example is a good point, since the final tracklist to use is ambiguous in that case. Agreed that an error makes sense there.

Though the case where it's just two releases being merged seems to have clear enough intent and is unambiguous to me. Maybe it's still inconsistent somehow? The error message just isn't very satisfying. :)

**Comment by reosarevok** [comment]
I guess the main benefit of this way is that it just works as is, while the alternative would require a lot of extra changes ;p

**Comment by reosarevok** [comment]
Changed the message a bit, added a test.

**Review by mwiencek — APPROVED** [review]

File: lib/MusicBrainz/Server/Data/Release.pm [hunk 1]
```diff
@@ -898,9 +899,23 @@ sub can_merge {
             });
         }
 
+        my $merging_into_empty_medium = $self->sql->select_single_value(
+            "$mediums_query
+             WHERE s.track_count > 0
+             AND new_medium.track_count = 0
+             LIMIT 1",
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Data/Release.pm:906]: Might be a good opportunity for `<<~EOSQL` :)

**Pull request merged**